astropy/stats/biweight.py

```diff
@@ -0,0 +1,638 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This module contains functions for computing robust statistics using
+Tukey's biweight function.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+
+from .funcs import median_absolute_deviation
+
+
+__all__ = ['biweight_location', 'biweight_scale', 'biweight_midvariance',
+           'biweight_midcovariance', 'biweight_midcorrelation']
+
+
+def biweight_location(a, c=6.0, M=None, axis=None):
+    r"""
+    Compute the biweight location.
+
+    The biweight location is a robust statistic for determining the
+    central location of a distribution.  It is given by:
+
+    .. math::
+
+        \zeta_{biloc}= M + \frac{\Sigma_{|u_i|<1} \ (x_i - M) (1 - u_i^2)^2}
+            {\Sigma_{|u_i|<1} \ (1 - u_i^2)^2}
+
+    where :math:`x` is the input data, :math:`M` is the sample median
+    (or the input initial location guess) and :math:`u_i` is the
+    indicator function given by:
```

Comment:
Thanks for review this, @mirca!  I removed the text calling `u_i` an indicator function, although `u_i`  it does act as a indicator function because the summation is only for `u_i` > 1 (the equation is sometimes expressed with coefficients that are 0 or 1 based on `u_i`).